api/dbadapters/sqldatawarehouse.ts

```diff
@@ -0,0 +1,111 @@
+import { Credentials } from "@dataform/api/commands/credentials";
+import { DbAdapter, OnCancel } from "@dataform/api/dbadapters/index";
+import { dataform } from "@dataform/protos";
+
+const sql = require("mssql");
+
+interface IRequest {
+  query: (query: string) => Promise<{ rows: any[] }>;
+}
+
+interface ISQLDataWarehouseConfig {
+  server?: string;
+  port?: number;
+  user?: string;
+  password?: string;
+  database?: string;
+  encrypt: boolean;
+}
+
+export class SQLDataWarehouseDBAdapter implements DbAdapter {
+  private pool: any;
+
+  constructor(credentials: Credentials) {
+    const sqlDataWarehouseCredentials = credentials as dataform.ISQLDataWarehouse;
+    const config: ISQLDataWarehouseConfig = {
+      server: sqlDataWarehouseCredentials.server,
+      port: sqlDataWarehouseCredentials.port,
+      user: sqlDataWarehouseCredentials.username,
+      password: sqlDataWarehouseCredentials.password,
+      database: sqlDataWarehouseCredentials.databaseName,
+      encrypt: true
+    };
+    const conn = new sql.ConnectionPool(config).connect();
+
+    conn.then(pool =>{
```

Comment:
it isn't, since I'm just throwing the error, the recommendation from the library though is to always attach an error listener: https://www.npmjs.com/package/mssql#connections-1

Reply:
thanks for the link!

Reply:
taking a slightly deeper look... there is a bit of a race here, i believe. if `execute` is called before this code is, no error handler will be attached before some real code is run.

i think we want to do something like:

this.pool = new Promise((resolve) => {
  const conn = new sql.ConnectionPool(config).connect();
  conn.then(pool => {
    pool.on('error', err => {
      throw new Error(err);
    });
    resolve(conn);
  });
});

Reply:
but, in the future if you decide to handle errors in the db adapters, it would just be clear where's the place to do it.

Reply:
got it (race condition), fixed.